### DISCUSSION TITLE
graphql/envelop/plugins/use-schema-by-context

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/envelop#1627] [Docs Discussions]
# graphql/envelop/plugins/use-schema-by-context

GraphQL Envelop Documentation

https://the-guild.dev/graphql/envelop/plugins/use-schema-by-context

**Comment by beklerken** [comment]
I wonder what kind of advantage this might have. So let's say we have an e-commerce site, after all, what kind of advantage will this be an index and lazy load?

**Reply by n1ru4l** [reply]
This plugin is useful if you want to serve a different schema based on the viewer. E.g. you could have one schema that is served for your public API and another schema that is served for you in-house web app. E.g. the public schema might only be a subset of the public schema.
